homeassistant/components/nest/config_flow.py

```diff
@@ -218,14 +218,9 @@ async def async_oauth_create_entry(self, data: dict[str, Any]) -> FlowResult:
             return await self.async_step_finish()
         return await self.async_step_pubsub()
 
-    async def async_step_reauth(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
```

Comment:
@allenporter the data in async_step_reauth should never be None, as it automatically passes in the config_entry data.
Had you actually encountered this scenario, or was it simply based on a bad assumption?

Reply:
It is a leftover from the old days as it seems; in the early days, it was possible.